tools/arcilator/arcilator.cpp

```diff
@@ -461,9 +461,11 @@ static LogicalResult processBuffer(
 
     mlir::ExecutionEngineOptions engineOptions;
     engineOptions.jitCodeGenOptLevel = llvm::CodeGenOptLevel::Aggressive;
-    engineOptions.transformer = mlir::makeOptimizingTransformer(
-        /*optLevel=*/3, /*sizeLevel=*/0,
-        /*targetMachine=*/nullptr);
+    static std::function<llvm::Error(llvm::Module *)> transformer =
```

Comment:
otherwise 

```
warning: object backing the pointer engineOptions.transformer will be destroyed at the end of the full-expression [-Wdangling-assignment-gsl]
```

because `engineOptions.transformer` is a `function_ref`.

Reply:
Is using a global variable the approved way of doing this upstream? I suspect a `auto transformer = ...; engineOptions.transformer = transformer;` might suffice 🤔

Reply:
it's locally allocated/scope - it'll die at the end of the body while the `executionEngine` is returned

Reply:
To me it looks like an `auto transformer = ...` would have the same lifetime as the execution engine. Where is the engine returned?

Reply:
oh um whoops - I skimmed the file too quickly (the return is `success`). Lemme change.